queries.sql

```diff
@@ -6,3 +6,101 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT *FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+
+-- Update all animals to have species as "unspecified"
+UPDATE animals
+SET species = 'unspecified';
+
+-- Verify the change
+SELECT * FROM animals;
+
+-- Roll back the change
+ROLLBACK;
+
+-- Verify the species column reverted to its previous state
+SELECT * FROM animals;
+
+BEGIN;
+
+-- Update animals with names ending in "mon" to have species as "digimon"
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+-- Update animals without a species already set to have species as "pokemon"
+UPDATE animals
+SET species = 'pokemon'
+WHERE species = '';
+
+-- Verify the changes
+SELECT * FROM animals;
+
+COMMIT;
+
+-- Verify the changes persist after the commit
+SELECT * FROM animals;
+
+-- Start the transaction
+BEGIN TRANSACTION;
+
+-- Delete all records in the animals table
+DELETE FROM animals;
+
+-- Rollback the transaction
+ROLLBACK;
+
+-- Verify if all records in the animals table still exist
+SELECT * FROM animals;
+
+-- Start the transaction
+BEGIN TRANSACTION;
+
+-- Delete all animals born after Jan 1st, 2022
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+-- Create a savepoint for the transaction
+SAVEPOINT my_savepoint;
+
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+-- Rollback to the savepoint
+ROLLBACK TO SAVEPOINT my_savepoint;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+-- Commit the transaction
+COMMIT;
+
+SELECT COUNT(*) AS total_animals
+FROM animals;
+
+SELECT COUNT(*) AS never_escaped_count
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight
+FROM animals;
+
+SELECT neutered, COUNT(*) AS escape_count
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC
+LIMIT 1;
+
+SELECT name, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY name;
```

Comment:
I think that since we have been asked to find the minimum and maximum weight of each type of animal we should instead use the species instead of the name 👍. The code will probably end up looking something like this: 

```sh
   SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight 
   FROM animals
   GROUP BY species;
```